src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -5,345 +5,132 @@
+                                                     m_batch_size);
+        set_tensor(it, res);
     }
 }
-void SyncInferRequest::SetBlobsToAnotherRequest(InferenceEngine::SoIInferRequestInternal& req) {
-    for (const auto& it : _networkInputs) {
-        auto& name = it.first;
+
+void SyncInferRequest::set_tensors_to_another_request(std::shared_ptr<ov::IAsyncInferRequest>& req) {
+    for (const auto& it : get_inputs()) {
         // this request is already in BUSY state, so using the internal functions safely
-        auto blob = GetBlob(name);
-        if (req->GetBlob(name) != blob)
-            req->SetBlob(name, blob);
+        auto tensor = get_tensor(it);
+        auto type = tensor.get_element_type();
+        if (req->get_tensor(it).data(type) != tensor.data(type)) {
+            req->set_tensor(it, ov::Tensor(it, tensor.data()));
+        }
     }
-    for (const auto& it : _networkOutputs) {
-        auto& name = it.first;
+    for (const auto& it : get_outputs()) {
         // this request is already in BUSY state, so using the internal functions safely
-        auto blob = GetBlob(name);
-        if (req->GetBlob(name) != blob)
-            req->SetBlob(name, blob);
+        auto tensor = get_tensor(it);
+        auto type = tensor.get_element_type();
+        if (req->get_tensor(it).data(type) != tensor.data(type)) {
+            req->set_tensor(it, ov::Tensor(it, tensor.data()));
+        }
     }
 }
 
-void SyncInferRequest::CopyInputsIfNeeded() {
-    for (const auto& it : _networkInputs) {
-        auto& name = it.first;
+void SyncInferRequest::copy_inputs_if_needed() {
+    for (const auto& it : get_inputs()) {
         // this request is already in BUSY state, so using the internal functions safely
-        CopyBlobIfNeeded(GetBlob(name), m_batched_request_wrapper._inferRequestBatched->GetBlob(name), true);
+        auto dst_tensor = m_batched_request_wrapper->_infer_request_batched->get_tensor(it);
+        copy_tensor_if_needed(get_tensor(it), dst_tensor, true);
     }
 }
 
-void SyncInferRequest::CopyBlobIfNeeded(InferenceEngine::Blob::CPtr src, InferenceEngine::Blob::Ptr dst, bool bInput) {
-    auto bufferDst = dst->buffer();
-    auto ptrDst = bufferDst.as<char*>();
-    auto bufferSrc = src->cbuffer();
-    auto ptrSrc = bufferSrc.as<const char*>();
-    ptrdiff_t szDst = dst->byteSize();
-    ptrdiff_t szSrc = src->byteSize();
+void SyncInferRequest::copy_tensor_if_needed(const ov::Tensor& src, ov::Tensor& dst, bool bInput) {
+    auto ptrDst = static_cast<char*>(dst.data());
+    auto ptrSrc = static_cast<char*>(src.data());
+    ptrdiff_t szDst = dst.get_byte_size();
+    ptrdiff_t szSrc = src.get_byte_size();
     if (bInput) {
         ptrdiff_t offset = szSrc != szDst ? m_batch_id * szDst / m_batch_size : 0;
-        if ((ptrDst + offset) == ptrSrc)
+        if ((ptrDst + offset) == ptrSrc) {
             return;
-        else
+        } else
             memcpy(ptrDst + offset, ptrSrc, szSrc);
```

Comment:
Do you consider applying ov::parallel_for() to improve memcpy's performance?

Reply:
I think we can put this optimization in next step.